core/src/main/java/org/apache/accumulo/core/client/impl/TabletServerBatchWriter.java

```diff
@@ -638,6 +648,21 @@ public MutationWriter(int numSendThreads) {
       queued = new HashSet<String>();
       sendThreadPool = new SimpleThreadPool(numSendThreads, this.getClass().getName());
       locators = new HashMap<String,TabletLocator>();
+      binTimer.schedule(new TimerTask() {
+        @Override
+        public void run() {
+          MutationSet m = queue.poll();
+          while (null != m) {
+            try {
+              addMutations(m);
+            } catch (Exception e) {
+              updateUnknownErrors("Error processing mutation set", e);
+            }
+            m = queue.poll();
+          }
+        }
+      }, 0, 500);
```

Comment:
Well, in the previous revision I was adding a MutationSet to an ExecutorService that had an unbounded queue and 1 thread. That's not immediate. Can MutationSets be binned concurrently? If so, then I can change the timer from a fixed delay to fixed rate. Thoughts?